classes/checkout/CheckoutAddressesStep.php

```diff
@@ -302,4 +302,20 @@ public function render(array $extraParams = array())
             $this->getTemplateParameters()
         );
     }
+    
+    public function setComplete($step_is_complete)
+    {
+        if ($step_is_complete && $this->context->cart->id_carrier == 0) {
+            $deliveryOptionSelected = $this->getCheckoutSession()->getSelectedDeliveryOption();
+            $id_address = $this->context->cart->id_address_delivery;
```

Comment:
Could you move the `$id_address` variable in the second if?

Reply:
It is too complicated to read now, isn't it?

Reply:
Yes it is. Sorry I misspoke.
I mean 

```
if ($step_is_complete && ....) {
  $id_address = ...;
  $this->getCheckoutSession()->setDeliveryOption(
  ...
}
```

Reply:
You wrote cleary and I get you right. But after moving the assigment of $id_address it became obvios that one if could be used. :)
I'm redoing.